solidity/contracts/Mailbox.sol

```diff
@@ -122,7 +147,13 @@ contract Mailbox is
         // Insert the message ID into the merkle tree.
         bytes32 _id = _message.id();
         tree.insert(_id);
-        emit Dispatch(_id, _message);
+        emit Dispatch(
+            msg.sender,
+            _destinationDomain,
+            _recipientAddress,
+            _message
+        );
+        emit DispatchId(_id);
```

Comment:
nit: Can we have the ID events be either both before or both after the non-ID events? No preference so long as they're the same order